arelle/plugin/validate/ESEF/__init__.py

```diff
@@ -121,21 +117,27 @@ def modelDocumentPullLoader(
     ) -> ModelDocumentClass | LoadingException | None:
         if not esefDisclosureSystemSelected(modelXbrl):
             return None
+        disclosureSystemYear = getDisclosureSystemYear(modelXbrl)
         if isEntry:
             if any("unconsolidated" in n for n in modelXbrl.modelManager.disclosureSystem.names):
-                if not is2021DisclosureSystem(modelXbrl) and re.match(r'.*[.](7z|rar|tar|jar)', normalizedUri):
+                if disclosureSystemYear > 2021 and re.match(r'.*[.](7z|rar|tar|jar)', normalizedUri):
                     modelXbrl.error("ESEF.Arelle.InvalidSubmissionFormat",
                                     _("Unrecognized submission format."),
                                     modelObject=modelXbrl)
                     return LoadingException("Invalid submission format")
             else:
-                if modelXbrl.fileSource.isArchive:
-                    if (not isinstance(modelXbrl.fileSource.selection, list) and
-                        modelXbrl.fileSource.selection is not None and
-                        modelXbrl.fileSource.selection.endswith(".xml") and
-                        ModelDocument.Type.identify(modelXbrl.fileSource, cast(str, modelXbrl.fileSource.url)) in (
-                            ModelDocument.Type.TESTCASESINDEX, ModelDocument.Type.TESTCASE)):
+                if isinstance(modelXbrl.fileSource.url, str) and modelXbrl.fileSource.url.endswith(".xml"):
+                    documentType = ModelDocument.Type.identify(modelXbrl.fileSource, modelXbrl.fileSource.url)
+                    if documentType in {ModelDocument.Type.TESTCASESINDEX, ModelDocument.Type.TESTCASE}:
                         return None  # allow zipped test case to load normally
+
+                if disclosureSystemYear >= 2023 and not modelXbrl.fileSource.isZip:
+                    modelXbrl.error("ESEF.2.6.1.disallowedReportPackageFileExtension",
+                                    _("A report package MUST conform to the .ZIP File Format Specification and MUST have a .zip extension."),
```

Comment:
Out of curiosity, where does the "MUST have a .zip extension" come from?

Reply:
Thanks.  Interesting that it's not mentioned by the filing manual...

Reply:
Well, in the ESEF reporting manual, the only reference I see is this one (2.6.1):

> ESMA recommends that issuers follow the recommendations of XBRL International,  which indicates how Inline XBRL documents are to be included within a taxonomy package.

The specifications of a taxonomy package extension are at

Reply:
2.6.3 speaks to it a bit, but doesn't describe it as a requirement like the conformance suite does:
> Whilst ESMA did not define in the RTS on ESEF a unique naming convention for ESEF
files, unless the relevant Officially Appointed Mechanism and / or National Competent
Authorities provide indications of any specific naming conventions which are required
at national level, ESMA encourages issuers to adopt a naming convention which match
{base}-{date}-{lang}.zip, whereby:
> ...
> The above naming convention is recommended both for the report package files (with .zip extension) and for any report file (with .html or .xhtml extension) present within the package. For the naming convention of the taxonomy files that are part of the report package, please refer to Guidance 3.1.5.